libs/agno/agno/os/routers/session/session.py

```diff
@@ -410,21 +554,38 @@ async def get_session_runs(
         if not runs:
             raise HTTPException(status_code=404, detail=f"Session with ID {session_id} has no runs")
 
+        # Filter runs by timestamp if specified
+        filtered_runs = []
+        for run in runs:
+            if start_timestamp or end_timestamp:
+                run_created_at = run.get("created_at")
+                if run_created_at:
+                    # created_at is stored as epoch int
+                    if start_timestamp and run_created_at < start_timestamp:
+                        continue
+                    if end_timestamp and run_created_at > end_timestamp:
+                        continue
```

Comment:
Could be expensive, ideally we do this at db level where possible. Let's ticket / add coment?